### PR TITLE
Implement WixSilentExec, and add new entry point for CAQuietExec (WixQuietExec)

### PR DESCRIPTION
**Pull request by Mike-GC**

### CODE REVIEW

File: src/chm/documents/customactions/qtexec.html.md [hunk 1]
```diff
@@ -23,14 +23,29 @@ When the QtExec action is run as an immediate custom action, it will try to exec
...
 
-If you want to run more than one command line in the immediate sequence then you will need to schedule multiple QtExec custom actions and set the QtExecCmdLine property to a new value by scheduling a property-setting custom action immediately before each instance of the QtExec custom action.
+If you want to run more than one command line in the immediate sequence then you will need to schedule multiple QtExec custom actions and set the WixQuietExecCmdLine property to a new value by scheduling a property-setting custom action immediately before each instance of the QtExec custom action.
+
+## Silent execution
+
+If you need to run a program without logging any of the input parameters or output of the executable, for example for security or privacy reasons, you want WixSilentExec:
```
Comment (barnson) [inline src/chm/documents/customactions/qtexec.html.md]: comma splice. You can put the whole "for example" phrase in --- em dashes. And then put a comma after "for example".

File: src/chm/documents/customactions/qtexec.html.md [hunk 2]
```diff
@@ -23,14 +23,29 @@ When the QtExec action is run as an immediate custom action, it will try to exec
...
+    .
+    .
+    .
+    <InstallExecuteSequence>
+      <Custom Action="SilentExecExample" After="TheActionYouWantItAfter"/>
+    </InstallExecuteSequence>
+
+The *only* difference in behavior between WixQuietExec and WixSilentExec is that WixSilentExec never logs the input or output of the commandline. Take special note to mark the input property and other properties as hidden if you do not want them logged automatically by MSI.
 
 ## Deferred execution
```
Comment (barnson) [inline src/chm/documents/customactions/qtexec.html.md:43]: Mention WixSilentExec in this section. Can be simple "(or WixSilentExec)" parenthetical.

File: src/chm/documents/customactions/qtexec.html.md [hunk 3]
```diff
@@ -55,16 +70,16 @@ If you need to set a command line that uses other Windows Installer properties,
 
 ## Running 64-bit executables
```
Comment (barnson) [inline src/chm/documents/customactions/qtexec.html.md:71]: As above for WixSilentExec64.

File: src/ext/ca/wixca/dll/qtexecca.cpp [hunk 4]
```diff
@@ -62,17 +94,17 @@ HRESULT BuildCommandLine(
...
-DWORD GetTimeout()
+DWORD GetTimeout(LPCSTR szPropertyName, LPCWSTR wzPropertyName)
 {
     DWORD dwTimeout = ONEMINUTE;
     HRESULT hr = S_OK;
 
     LPWSTR pwzData = NULL;
 
-    if (WcaIsPropertySet("QtExecCmdTimeout"))
+    if (WcaIsPropertySet(szPropertyName))
```
Comment (barnson) [inline src/ext/ca/wixca/dll/qtexecca.cpp]: Will WcaIsUnicodePropertySet work here instead? I :dizzy_face: at the duplication.

File: src/chm/documents/customactions/qtexec.html.md (hunk 1 above)
Comment (Mike-GC) [inline src/chm/documents/customactions/qtexec.html.md]: Resolved.

File: src/chm/documents/customactions/qtexec.html.md (hunk 2 above)
Comment (Mike-GC) [inline src/chm/documents/customactions/qtexec.html.md:43]: Resolved.

File: src/chm/documents/customactions/qtexec.html.md (hunk 3 above)
Comment (Mike-GC) [inline src/chm/documents/customactions/qtexec.html.md:71]: Resolved.

File: src/ext/ca/wixca/dll/qtexecca.cpp (hunk 4 above)
Comment (Mike-GC) [inline src/ext/ca/wixca/dll/qtexecca.cpp]: Good point, didn't think to check if we had such functionality. Resolved.

**Pull request merged**